subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -164,11 +162,12 @@ static void send_pending_adv(struct k_work *work)
 			continue;
 		}
 
-		BT_MESH_ADV(buf)->busy = 0U;
 		err = buf_send(buf);
 		if (!err) {
 			return; /* Wait for advertising to finish */
 		}
+
+		net_buf_unref(buf);
```

Comment:
Or is the intention that this above is the unref for `adv.buf`? In that case the reference management is rather careless. Either an API takes ownership of a reference or it doesn't. You can't have a mix of both. So if `buf_send()` took ownership (which it seems to do since you're not calling `ref()` there) then you have no right to touch `buf` anymore in this function. Perhaps you should be adding the `ref()` to `buf_send()` so that the caller retains ownership?